### ISSUE TITLE
Guarantee failed: [progress >= metainfo_threshold]

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4203]
After restarting a server in a 3-server cluster with a replicated table, I got this after a couple seconds:

```
error: Error in src/rdb_protocol/store_receive_backfill.cc at line 423:
error: Guarantee failed: [progress >= metainfo_threshold] 
error: Backtrace:
error: Thu May  7 20:43:51 2015
       
       1: format_backtrace(bool) at backtrace.cc:214
       2: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
       3: ./rethinkdb() [0xba5594] at 0xba5594 ()
       4: apply_single_key_item(receive_backfill_tokens_t const&, backfill_item_t&) at store_receive_backfill.cc:250
       5: coro_t::run() at coroutines.cc:214
error: Exiting.
Trace/breakpoint trap (core dumped)
```

This was in release mode, in branch `raft` at commit 0cfa4a948b4f451c88ed32fc4b30ff23a90852c1.

@timmaxw any ideas?

**Comment by danielmewes** [comment]
This seems to happen every time I restart the server.

Data files are on sinister in /home/daniel/4203-data. 
To reproduce, start the following servers:
```
rethinkdb -o 1 -d data_1 -j localhost &
rethinkdb -o 2 -d data_2 -j localhost &
rethinkdb -d data_0
```

The server on data_0 will crash after a moment.

**Comment by timmaxw** [comment]
Fixed in 28bad166792384316311e856c57e9aab66d4dd96.
